pkg/build/builder/cmd/builder.go

```diff
@@ -18,27 +18,23 @@ import (
 	buildapiv1 "github.com/openshift/api/build/v1"
 	buildclientv1 "github.com/openshift/client-go/build/clientset/versioned/typed/build/v1"
 	"github.com/openshift/library-go/pkg/git"
-	"github.com/openshift/origin/pkg/api/legacy"
 	bld "github.com/openshift/origin/pkg/build/builder"
 	"github.com/openshift/origin/pkg/build/builder/cmd/scmauth"
 	"github.com/openshift/origin/pkg/build/builder/timing"
 	builderutil "github.com/openshift/origin/pkg/build/builder/util"
-	buildutil "github.com/openshift/origin/pkg/build/util"
 	s2iapi "github.com/openshift/source-to-image/pkg/api"
 	s2igit "github.com/openshift/source-to-image/pkg/scm/git"
 )
 
 var (
-	buildEnvVarScheme       = runtime.NewScheme()
-	buildEnvVarCodecFactory = serializer.NewCodecFactory(buildEnvVarScheme)
-	buildEnvVarJSONCodec    runtime.Codec
+	buildScheme       = runtime.NewScheme()
+	buildCodecFactory = serializer.NewCodecFactory(buildScheme)
+	buildJSONCodec    runtime.Codec
 )
 
 func init() {
-	// TODO only use external versions, so we only add external types
-	legacy.InstallInternalLegacyBuild(buildEnvVarScheme)
-	utilruntime.Must(buildapiv1.AddToScheme(buildEnvVarScheme))
-	buildEnvVarJSONCodec = buildEnvVarCodecFactory.LegacyCodec(buildapiv1.SchemeGroupVersion, legacy.GroupVersion)
+	utilruntime.Must(buildapiv1.AddToScheme(buildScheme))
```

Comment:
Wait, why not just use the scheme in the build client? `github.com/openshift/client-go/build/clientset/versioned/scheme`

Reply:
Then you don't need this or to define your own scheme.